### PR TITLE
Add queue percentage to libbeat metrics

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## Proposed commit message

Part of https://github.com/elastic/beats/issues/38708

This adds a `queue.full` metric that reports the percentage of queue usage in libbeat.

I'm not sure if this is the _correct_ way to measure queue usage in percentage, but this was such a small change I figured it would be faster to just put in a PR and ask, rather than ask and wait.

## Testing

You can test this by starting metricbeat with `--httpprof localhost:9898` and then checking the metrics with `curl localhost:9898/debug/vars | grep libbeat`

The metric will also appear in the `last 30s` metrics.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Review by belimawr — COMMENTED** [review]
LGTM

Now that we can know the queue size in bytes, I'm just wondering if we should make it more clear that the percentage is calculated from the number of events.

**Review by belimawr — APPROVED** [review]

**Comment by cmacknz** [comment]
I would update the integration as part of this work. If you can't for some reason, create a separate issue to do it so it isn't lost.

**Comment by cmacknz** [comment]
> Now that we can know the queue size in bytes, I'm just wondering if we should make it more clear that the percentage is calculated from the number of events.

The queue size limit is still configured in units of events and the metric here should match the units of the maximum size limit.

**Review by faec — APPROVED** [review]

**Comment by fearful-symmetry** [comment]
Huh, it's possible for us to `activeEvents` > `max_events`:
```
{"clients":2,"events":{"active":3201,"published":99200,"total":99200},"queue":{"acked":99200,"filled":{"pct":{"events":1.0003125}},"max_events":3200}}
```

This happens occasionally, in the metrics, and it's always max_queue+1. I assume it's just a result of the queue itself not being completely synced up with the metrics reporter. That, or an off-by-one. @faec you seen this before?

**Comment by faec** [comment]
> This happens occasionally, in the metrics, and it's always max_queue+1. I assume it's just a result of the queue itself not being completely synced up with the metrics reporter. That, or an off-by-one. @faec you seen this before?

Yeah, I've seen this, it's just an artifact of the metrics receiver getting notified of acks slightly after the queue unblocks so it sees the new event in flight before decrementing the old ones. I've never seen the `pipeline` report something that's actually wrong, just slightly behind on acks, though I did see recently that the `output` subtree sometimes has a number that's just flat-out wrong https://github.com/elastic/beats/issues/39146

File: libbeat/publisher/pipeline/monitoring.go [hunk 1]
```diff
@@ -121,12 +123,23 @@ func (o *metricsObserver) clientClosed() { o.vars.clients.Dec() }
...
 	o.vars.events.Inc()
 	o.vars.activeEvents.Inc()
+	o.setPercentageFull()
+}
+
+// setPercentageFull is used interally to set the `queue.full` metric
+func (o *metricsObserver) setPercentageFull() {
+	maxEvt := o.vars.queueMaxEvents.Get()
+	if maxEvt != 0 {
+		pct := float64(o.vars.activeEvents.Get()) / float64(maxEvt)
```
Comment (cmacknz) [inline libbeat/publisher/pipeline/monitoring.go:133]: You should limit the precision so we aren't including arbitrary insignificant digits. 

You could also report this as an integer between 0-100 to eliminate this problem.

**Comment by fearful-symmetry** [comment]
Alright, added rounding. I decided to keep the percent calculations "raw" (0.0-1.0, not 0.0-100.0), since that's what we do for the system metrics percentages, and I figure we might as well keep it consistent?

**Pull request merged**